### ISSUE TITLE
Missing heterogenous deletion

### ISSUE THREAD

**Issue by chapmanb** [AstraZeneca-NGS/VarDictJava#26]
@zhongwulai and all;
@ohofmann ran into a variant calling case where VarDict appears to be missing a real indel. Running FreeBayes on the same region results in a deletion call with high depth and a clear separation between tumor (first sample) and normal (second):
```
17      29562656        .       CTGTTT  CT      2102.99 .       AB=0.337979;ABP=68.4496;AC=1;AF=0.25;AN=4;AO=97;CIGAR=1M4D1M;DP=322;DPB=275.833;DPRA=8.2;EPP=6.79359;EPPR=3.49176;GTI=0;LEN=4;MEANALT=4;MQM=60;MQMR=60;NS=2;NUMALT=1;ODDS=48.4
244;PAIRED=1;PAIREDR=0.99095;PAO=9;PQA=224.5;PQR=1014.5;PRO=32;QA=3480;QR=8273;RO=221;RPL=49;RPP=3.03269;RPPR=3.80618;RPR=48;RUN=1;SAF=43;SAP=5.71904;SAR=54;SRF=119;SRP=5.84992;SRR=102;TYPE=del;technology.illumina=1   GT:DP:RO:QR:AO:QA:GL    0/1:287:187:7007:97:3480:-218.303,0,-600.114    0/0:35:34:1266:0:0:0,-10.8371,-120.597
```
The read support looks reasonable in terms of strand and positioning, so I couldn't identify an obvious reason why VarDict skips it. The VarDict output has no variants listed at all so I'm a big unsure how best to debug. Here is a minimal example and shell script that demonstrate the issue:

https://s3.amazonaws.com/chapmanb/vardict_missing_indel.tar.gz

I'd welcome any thoughts on digging into this to improve sensitivity for this case.

**Comment by msheimlich** [comment]
Hi Brad and Zhongwu.

Would you know if there is a difference between the Java and Perl versions
missing this?

Thanks.
Michael

**Comment by chapmanb** [comment]
Michael -- the Java and Perl versions are the same. Both don't return any output on the test case. Thanks much for looking at this.

**Comment by zhongwulai** [comment]
Hi Oliver and Brad,

Thanks for the feedback and example.

The deletion can be called in single sample mode, but not called in paired mode.

I dig into a bit, and found that it's filtered out in the combined analysis step.  This step is meant to identify potential false positives for Indels. In this step, two bam files are artificially merged and variants is called as if it's a single sample.  If the same indel is not called, or is called but produced less supporting reads, it's deemed false positives and filtered.  In this case, it's the latter one.  Combined analysis produces 111 reads while single sample produces 115 reads (-y option and the info is printed in the last line "Combine produce less: 17 29562657 -4 111 115").  This is the second time it's reported, and it appears to be too strict.  The reason it produce less in combined step is there's microhomology and VarDict tries to recover from short overhangs in tumor samples.  This is over-written in combined analysis as now there're many reads spanning the entire gap.

The simplest fix is to change line #497 (return "FALSE";) to (return "";) in Perl version, and it should be called.  The caveat is it might produce more false positives, but everything is a trade off.

In Java version, change line #895 in VarDict.java from (return tuple(rlen, "FALSE");) to (return tuple(rlen, "");), and it should fix the problem.

Thanks,
Zhongwu

**Comment by chapmanb** [comment]
Zhongwu;
Thanks for the detailed summary, that helps a lot. I can see how this could trigger an issue for het indels that are not present in the normal at all, since any changes would result in the indel being lost. If it provides value in some cases where you're filtering, I wonder if it would make sense to have this be a less strict cutoff to not remove it entirely. Right now you have `vref.cov < var1.cov - 2`, What do you think about triggering it if you lose, say, more than 25% of the total reads in the merged sample, so `vref.cov / var1.cov > .75`? I know this is just introducing a heuristic but it seems to capture the spirit of your initial implementation without losing calls in these cases where it gets slightly reduced.

**Comment by chapmanb** [comment]
Fixed with updates in #7
